SerialAPI.h

```diff
@@ -87,6 +89,25 @@ class SerialAPI: public FB::JSAPIAuto
 			delete [] lpCC;
 		}
 #endif
+#ifdef __unix__
+            DIR *dev = opendir("/dev");
+            struct dirent *ent;
+
+            if (dev != NULL) {
+                while (ent = readdir(dev)) {
+                    std::string name(ent->d_name);
+                    std::string prefix = name.substr(0, 6); 

```

Comment:
From http://www.cplusplus.com/reference/string/string/substr/
"if the string is shorter, as many characters as possible are used"